src/stores/spaces/SpaceStore.ts

```diff
@@ -107,6 +107,12 @@ export class SpaceStoreClass extends AsyncStoreWithClient<IState> {
     private childSpacesBySpace: SpaceDescendantMap = new Map<SpaceKey, Set<SpaceKey>>();
     // Map from SpaceKey to Set of user IDs that are direct descendants of that space
     private userIdsBySpace: SpaceEntityMap = new Map<Room["roomId"], Set<string>>();
+    // cache that stores the aggregated lists of roomIdsBySpace and userIdsBySpace
+    // cleared on changes
+    private _aggregatedSpaceCache: Record<string, SpaceEntityMap> = {
```

Comment:
this type is weird, given the keys are not plain strings, they're just a couple of specific explicit fields, imo would be better without an explicit type defn